include/arch/aarch32/el3_common_macros.S

```diff
@@ -94,7 +94,7 @@
 	 *  from all exception levels.
 	 * ---------------------------------------------------------------------
 	 */
-#if (ARM_ARCH_MAJOR > 7) || defined(ARMV7_SUPPORTS_VFP)
+#if ((ARM_ARCH_MAJOR > 7) || defined(ARMV7_SUPPORTS_VFP)) && !(__SOFTFP__)
```

Comment:
AIU, ARMV7_SUPPORTS_VFP needs hard-float toolchain.  Enabling this build flag without hard-float should be disallowed. Hence would be better to prevent setting this flag without HF compiler would be a better approach.

Reply:
Your suggestion will work with V7 cores like A7/A15 etc, but for Cortex-A32 we still need to make decision based on HF/SF toolchain because ARCH_MAJOR for Cortex-A32 is 8.

Reply:
The soft-float/hard-float distinction is only for ARMv7. ARMv8 is always hard-float. The soft-float compiler should not be used.

Reply:
The problem is with Cortex-A32 which is unique in a sense that even though it is ARMv8 but it uses "arm-linux-gnueabi" toolchain which has both soft/hard-float variants and so there could be scenarios where soft-float toolchain may be used to compile it.
Real example of this scenario is,  we use Yocto to build our SW stack and the default Linaro toolchain, when aarch32 is used, is sof-float toolchain and because of that build fails.

Reply:
OK, all this is information should added to commit message and code comments.